demo/agw/ScrolledThumbnail.py

```diff
@@ -0,0 +1,640 @@
+        help_menu.Append(AS_ABOUT, "&About...")
+        self.Bind(wx.EVT_MENU, self.OnAbout, id=AS_ABOUT)
+
+        menu_bar = wx.MenuBar()
+
+        menu_bar.Append(file_menu, "&File")
+        menu_bar.Append(help_menu, "&Help")
+
+        return menu_bar
+
+
+    def OnClose(self, event):
+
+        self.Destroy()
+
+
+    def OnAbout(self, event):
+
+        msg = "This Is The About Dialog Of The ScrollThumbnail Demo.\n\n" + \
+              "Author: Michael Eager @ 23 Oct 2020\n\n" + \
+              "Adapted from the ThumbnailCtrl Demo\n" + \
+              "By Andrea Gavana @ 10 Dec 2005\n\n" + \
+              "Please Report Any Bug/Requests Of Improvements\n" + \
+              "To Me At The Following Addresses:\n" + \
+              "eager@eagercon.com\n\n" + \
+              "Welcome To wxPython " + wx.VERSION_STRING + "!!"
+
+        dlg = wx.MessageDialog(self, msg, "ScrolledThumbnail Demo",
+                               wx.OK | wx.ICON_INFORMATION)
+
+        dlg.SetFont(wx.Font(8, wx.NORMAL, wx.NORMAL, wx.NORMAL, False))
+        dlg.ShowModal()
+        dlg.Destroy()
+
+
+    def OnSetDirectory(self, event):
+
+        dlg = wx.DirDialog(self, "Choose a directory with images:",
+                           defaultPath=os.getcwd(),
+                           style=wx.DD_DEFAULT_STYLE|wx.DD_NEW_DIR_BUTTON)
+
+        # If the user selects OK, then we process the dialog's data.
+        # This is done by getting the path data from the dialog - BEFORE
+        # we destroy it.
+        if dlg.ShowModal() == wx.ID_OK:
+            self.ShowDir(dlg.GetPath())
+            self.log.write("OnSetDirectory: directory changed to: %s\n"%dlg.GetPath())
+
+        # Only destroy a dialog after you're done with it.
+        dlg.Destroy()
+
+
+    def ShowDir(self, dir):
+        files = os.listdir(dir)
+        thumbs = []
+        for f in files:
+            if os.path.splitext(f)[1] in [".jpg", ".gif", ".png"]:
+                statinfo = os.stat(os.path.join(dir, f))
+                size = statinfo.st_size
+                modtime = statinfo.st_mtime
+                TIME_FMT = '%d %b %Y, %H:%M:%S'
+                lastmod = time.strftime(TIME_FMT, time.localtime(modtime))
+                thumbs.append(Thumb(dir, f, caption=f, size=size, lastmod=lastmod,
+                                    imagehandler=NativeImageHandler))
+        self.scroll.ShowThumbs(thumbs)
+
+
+    def OnChangeOutline(self, event): # wxGlade: MyFrame.<event_handler>
+
+        radio = event.GetEventObject()
+        pos = self.radios.index(radio)
+
+        if pos == 0:
+            self.scroll.SetThumbOutline(self.scroll.THUMB_OUTLINE_NONE)
+        elif pos == 1:
+            self.scroll.SetThumbOutline(self.scroll.THUMB_OUTLINE_FULL)
+        elif pos == 2:
+            self.scroll.SetThumbOutline(self.scroll.THUMB_OUTLINE_RECT)
+        elif pos == 3:
+            self.scroll.SetThumbOutline(self.scroll.THUMB_OUTLINE_IMAGE)
```

Comment:
Playing around with the demo throws exceptions here like: 
```
Traceback (most recent call last):
  File "/Users/robind/projects/wx/git/Phoenix/demo/agw/ScrolledThumbnail.py", line 253, in OnChangeOutline
    self.scroll.SetThumbOutline(self.scroll.THUMB_OUTLINE_RECT)
AttributeError: 'ScrolledThumbnail' object has no attribute 'THUMB_OUTLINE_RECT'
```

Reply:
Fixed.